SignalMessaging/categories/UIColor+OWS.m

```diff
@@ -196,123 +225,311 @@ + (UIColor *)ows_blackColor
+            @"steel" : self.ows_steelShadeColor,
         };
+        OWSAssertDebug([self.ows_conversationColorMap.allKeys isEqualToArray:colorMap.allKeys]);
     });
 
     return colorMap;
 }
 
-+ (NSArray<NSString *> *)ows_conversationColorNames
++ (NSDictionary<NSString *, UIColor *> *)ows_conversationColorMapTint
 {
-    OWSAssertDebug(
-        [self.ows_conversationColorMapLight.allKeys isEqualToArray:self.ows_conversationColorMapDark.allKeys]);
+    static NSDictionary<NSString *, UIColor *> *colorMap;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        colorMap = @{
+            @"crimson" : self.ows_crimsonTintColor,
+            @"vermilion" : self.ows_vermilionTintColor,
+            @"burlap" : self.ows_burlapTintColor,
+            @"forest" : self.ows_forestTintColor,
+            @"wintergreen" : self.ows_wintergreenTintColor,
+            @"teal" : self.ows_tealTintColor,
+            @"blue" : self.ows_blueTintColor,
+            @"indigo" : self.ows_indigoTintColor,
+            @"violet" : self.ows_violetTintColor,
+            @"plum" : self.ows_plumTintColor,
+            @"taupe" : self.ows_taupeTintColor,
+            @"steel" : self.ows_steelTintColor,
+        };
+        OWSAssertDebug([self.ows_conversationColorMap.allKeys isEqualToArray:colorMap.allKeys]);
+    });
 
-    return self.ows_conversationColorMapLight.allKeys;
+    return colorMap;
 }
 
-+ (nullable UIColor *)ows_conversationColorForColorName:(NSString *)colorName isShaded:(BOOL)isShaded
++ (NSArray<NSString *> *)ows_conversationColorNames
 {
-    OWSAssertDebug(colorName.length > 0);
+    return self.ows_conversationColorMap.allKeys;
+}
 
-    return (isShaded ? self.ows_conversationColorMapDark : self.ows_conversationColorMapLight)[colorName];
++ (nullable OWSConversationColors *)ows_conversationColorsForColorName:(NSString *)conversationColorName
+{
+    UIColor *_Nullable defaultColor = self.ows_conversationColorMap[conversationColorName];
+    UIColor *_Nullable shadeColor = self.ows_conversationColorMapShade[conversationColorName];
+    UIColor *_Nullable tintColor = self.ows_conversationColorMapTint[conversationColorName];
+    if (!defaultColor || !shadeColor || !tintColor) {
+        return nil;
+    }
+    OWSAssertDebug(defaultColor);
+    OWSAssertDebug(shadeColor);
+    OWSAssertDebug(tintColor);
+    return [OWSConversationColors conversationColorsWithDefaultColor:defaultColor
+                                                          shadeColor:shadeColor
+                                                           tintColor:tintColor];
+}
+
++ (OWSConversationColors *)ows_conversationColorsOrDefaultForColorName:(NSString *)conversationColorName
+{
+    OWSConversationColors *_Nullable conversationColors =
+        [self ows_conversationColorsForColorName:conversationColorName];
+    if (conversationColors) {
+        return conversationColors;
+    }
+    return [self ows_defaultConversationColors];
```

Comment:
Like Android, I think we want to map legacy colors to their new color value here, otherwise won't we have a whole lot of defaultConversationColor'd avatars?

Reply:
This will be in a forthcoming PR.